src/Signup.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import axios from "axios";
+import history from './history';
+import './todo.css';
+
+class Signup extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            username: "",
+            password: "",
+            confirmPassword: "",
+            cred: {}
+        };
+    }
+
+    setUsername = event => {
+        let value = event.target.value;
+        this.setState({
+            username: value
+        });
+    };
+
+    setPassword = event => {
+        let value = event.target.value;
+        this.setState({
+            password: value
+        });
+    };
+
+    setConfirmPassword = event => {
+        let value = event.target.value;
+        this.setState({
+            confirmPassword: value
+        });
+    }
+
+    signUp = () => {
+        if (this.state.username === '' || this.state.password === '') {
+            alert('Fields can not be empty');
+        }
+        else if (this.state.password.length > 5) {
+            if (this.state.password === this.state.confirmPassword) {
+                axios({
+                    method: "post",
+                    url: `http://localhost:8080/addUser`,
+                    data: {
+                        username: this.state.username,
+                        password: this.state.password
+                    }
+                }).then((response) => {
+                    if (response) {
+                        alert("Registration successful");
+                        this.setState({
+                            username: '',
+                            password: '',
+                            confirmPassword: ''
+                        })
+                    }
+                }).catch(() => {
```

Comment:
this is not for the frontend to decide.The backend needs to send an appropirate HTTP code